test/foundry/AngstromRouter.t.sol

```diff
@@ -0,0 +1,200 @@
+        steps[0] = SwapPathStep({ pool: pool, tokenOut: usdc, isBuffer: false });
+        SwapPathExactAmountOut[] memory paths = new SwapPathExactAmountOut[](1);
+        paths[0] = SwapPathExactAmountOut({
+            tokenIn: dai,
+            steps: steps,
+            exactAmountOut: 1e18,
+            maxAmountIn: MAX_UINT256
+        });
+
+        uint256 snapId = vm.snapshot();
+        _prankStaticCall();
+        (
+            uint256[] memory pathAmountsInQuery,
+            address[] memory tokensInQuery,
+            uint256[] memory amountsInQuery
+        ) = angstromBalancer.querySwapExactOut(paths, bob, bytes(""));
+        vm.revertTo(snapId);
+
+        makeAngstromNode(bob);
+
+        vm.prank(bob);
+        (uint256[] memory pathAmountsIn, address[] memory tokensIn, uint256[] memory amountsIn) = angstromBalancer
+            .swapExactOut(paths, MAX_UINT256, false, bytes(""));
+
+        assertEq(pathAmountsIn.length, pathAmountsInQuery.length, "Path amounts in length is not equal");
+        assertEq(tokensIn.length, tokensInQuery.length, "Tokens in length is not equal");
+        assertEq(amountsIn.length, amountsInQuery.length, "Amounts in length is not equal");
+
+        for (uint256 i = 0; i < pathAmountsIn.length; i++) {
+            assertEq(pathAmountsIn[i], pathAmountsInQuery[i], "Path amounts in is not equal");
+            assertEq(tokensIn[i], tokensInQuery[i], "Tokens in is not equal");
+            assertEq(amountsIn[i], amountsInQuery[i], "Amounts in is not equal");
+        }
+    }
+
+    function _approveAngstromRouterForAllUsers() private {
+        for (uint256 i = 0; i < users.length; ++i) {
+            address user = users[i];
+            vm.startPrank(user);
+            approveAngstromRouterForSender();
+            vm.stopPrank();
+        }
+    }
+
+    function approveAngstromRouterForSender() internal {
+        for (uint256 i = 0; i < tokens.length; ++i) {
+            permit2.approve(address(tokens[i]), address(angstromBalancer), type(uint160).max, type(uint48).max);
+        }
+
+        for (uint256 i = 0; i < oddDecimalTokens.length; ++i) {
+            permit2.approve(
+                address(oddDecimalTokens[i]),
+                address(angstromBalancer),
+                type(uint160).max,
+                type(uint48).max
+            );
+        }
+
+        for (uint256 i = 0; i < erc4626Tokens.length; ++i) {
+            permit2.approve(address(erc4626Tokens[i]), address(angstromBalancer), type(uint160).max, type(uint48).max);
+        }
+    }
+
+    function approveAngstromRouterForPool(IERC20 bpt) internal {
+        for (uint256 i = 0; i < users.length; ++i) {
+            vm.startPrank(users[i]);
+            bpt.approve(address(angstromBalancer), type(uint256).max);
+            permit2.approve(address(bpt), address(angstromBalancer), type(uint160).max, type(uint48).max);
+            vm.stopPrank();
+        }
+    }
```

Comment:
Huh, interesting that we don't have these in the base vault test. 
Let's leave it for now, but eventually it could be a good addition to the base layer whenever we're testing custom routers.